python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -1204,6 +1192,119 @@ def _test_rotation_agreement(self, rot_vects, max_dist_rad):
                 tot_consent += 1
         return tot_consent
 
+    def _final_verify(self,
+                      source_array,
+                      shift_rot_matrix,
+                      max_dist_rad,
+                      min_matches):
+        """Match the all sources into the reference cat using the shift/rot
+        matrix.
+
+        After the initial set of matches and found to have enough matches,
+        refit the shift/rot matrix for a more stable solution.
+
+        Parameters
+        ----------
+        source_array : `numpy.ndarray` (N, 3)
+            3-vector positions on the unit-sphere representing the sources to
+            match
+        shift_rot_matrix : `numpy.ndarray` (3, 3)
+            Rotation matrix representing inferred shift/rotation of the
+            sources onto the reference catalog. Matrix need not be unitary.
+        max_dist_rad : `float`
+            Maximum distance allowed for a match.
+        min_matches : `int`
+            Minimum number of matched objects required to consider the
+            match good.
+
+        Returns
+        -------
+        output_struct : `lsst.pipe.base.Struct`
+            Result struct with components:
+
+            - ``match_ids`` : Pairs of indexes into the source and reference
+              data respectively defining a match (`numpy.ndarray`, (N, 2)).
+            - ``distances_rad`` : distances to between the matched objects in
+              the shift/rotated frame. (`numpy.ndarray`, (N,)).
+            - ``max_dist_rad`` : Value of the max matched distance. Either
+              returning the input value of the 2 sigma clipped value of the
+              shift/rotated distances. (`float`).
+        """
+        output_struct = pipeBase.Struct(
+            match_ids=None,
+            distances_rad=None,
+            max_dist_rad=None,
+        )
```

Comment:
It would be simpler to return None on failure rather than a struct full of Nones. That way you wouldn't have to generate this at the start, the short-circuit error returns would be more obvious (`return None` is clearer than `return output_struct`), and the test in `match()` would be much simpler.

Reply:
Pretty sure others had an opinion against just returning None when I first coded things this way. For consistency with the rest of the code I'm going to leave it as is.